core/new-gui/src/app/common/service/user/auth.interceptor.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from "@angular/core";
+import { HttpErrorResponse, HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from "@angular/common/http";
+import { Observable, throwError } from "rxjs";
+import { catchError } from "rxjs/operators";
+import { Router } from "@angular/router";
+
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+  constructor(private router: Router) {}
+  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    return next.handle(request).pipe(
+      catchError((err: unknown) => {
+        if (err instanceof HttpErrorResponse) {
+          if (err.status === 401) {
+            this.router.navigate(["home"]);
+          }
+        }
```

Comment:
This logic is triggered when JTW stored in the frontned is expired or invalid. Thus besides redirecting to home, shall we clean the stored JWT as well?

Reply:
It will be cleaned by the function registerAutoRefreshToken in auth.service. 

The JWT is already cleaned when we reach here.